provider/vsphere/internal/vsphereclient/createvm.go

```diff
@@ -106,30 +108,28 @@ type CreateVirtualMachineParams struct {
 	EnableDiskUUID bool
 }
 
-// CreateVirtualMachine creates and powers on a new VM.
+// vmTemplatePath returns the well-known path to
+// the template VM for this controller
 //
-// This method imports an OVF template using the vSphere API. This process
-// comprises the following steps:
-//   1. Ensure the VMDK contained within the OVA archive (args.OVA) is
-//      stored in the datastore, in this controller's cache. If it is
-//      there already, we use it; otherwise we remove any existing VMDK
-//      for the same series, and upload the new one.
-//   2. Call CreateImportSpec [0] with a pre-canned OVF, which validates
-//      the OVF descriptor against the hardware supported by the host system.
-//      If the validation succeeds,/the method returns an ImportSpec to use
-//      for importing the virtual machine.
-//   3. Prepare all necessary parameters (CPU, memory, root disk, etc.), and
-//      call the ImportVApp method [0]. This method is responsible for actually
-//      creating the VM. This VM is temporary, and used only to convert the
-//      VMDK file into a disk type file.
-//   4. Clone the temporary VM from step 3, to create the VM we will associate
-//      with the Juju machine.
-//   5. If the user specified a root-disk constraint, extend the VMDK if its
-//      capacity is less than the specified constraint.
-//   6. Power on the virtual machine.
+// Example:
+//   vmTemplatePath("juju-abc123-1")
+//   "juju-abc123-template"
+func vmTemplatePath(vmpath string) string {
+	parts := strings.Split(vmpath, "-")
+	templateNameParts := []string{
+		parts[0],
+		parts[1],
+		"template",
+	}
+	return strings.Join(templateNameParts, "-")
+}
+
+// CreateVirtualMachine creates and powers on a new VM.
```

Comment:
Should we keep the detailed explanation?

Reply:
Happy to add one, but I did want to make sure that we weren't keeping the stale comment around.

Reply:
Added a fuller comment in.